os/se/ameba/security_ameba_wrapper_tz.c

```diff
@@ -589,9 +592,11 @@ int se_ameba_hal_get_factory_key(uint32_t key_idx, hal_data *key)
 	AWRAP_ENTER;
 	int ret = HAL_SUCCESS;
 
+	device_mutex_lock(RT_DEV_LOCK_FLASH);
```

Comment:
Are you suggesting that we change the current Flash driver structure, so no flow control need for caller from above layer?
Is it suitable now?

Reply:
I mean your flow doesn't check flash status (Write enabled or write status) , so it never waits until previous write finished.
Instead of mutex, you should check status of flash.

Reply:
If others are operating Flash, mutex will be locked and only be unlocked when operation Flash is complete.
SE cannot acquire the mutex lock when it is locked, it will wait for unlock.
So there is no need to check write status.

Reply:
As Per my experience, even if write finished, sometimes h/w still writing on flash, so some of delay was necessary.
actually it was external flash, but are you sure your internal flash finished immediately after write / erase returns result??
Another our flash driver always check write status due to this behavior.

Reply:
We do have checks in our API, Flash Write API will wait for write enable bit before it returning result.

Reply:
That is what I ask to you about your flow actually =D
If your bsp already have such kind of logic, then every write time, it should check status before write.
```
func_write()
{
   write_on_flash;
   while(1) {
      is_wait_bit_true;
   }
   return result;
}
```
so above is your style, then below is my suggestion.
```
func_wait()
{
   while(1) {
      is_wait_bit_true;
   }
}
func_write;
{
   func_wait();
   write_on_flash;
   return result;
}
```
In this case, driver doesn't necessary mutex actually, caller should handle it.
But you should not fix it in this PR.

Reply:
Does this mean that this PR does not require further modification?